src/index.js

```diff
@@ -183,14 +183,64 @@ async function getCFPhysicalResourceId(key, awsParameters) {
   return result.StackResourceDetail
 }
 
+/**
+ * @param key the name of the APIGateway Api (Rest)
+ * @param awsParameters parameters to pass to the AWS.ApiGateway constructor
+ * @returns { Promise.<AWS.APIGateway.Api> }
+ * @see https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/APIGateway.html#getRestApis-property
+ */
+async function getAPIGatewayValue(key, awsParameters) {
```

Comment:
Can you pull some of the common functionality between these two methods out into a helper?

Reply:
Thanks for the review. 

I actually looked at this before, but it felt kinda awkward because the two APIGateway APIs have about as much in common as they have with e.g. RDS. Never hurts to have another look though, will let you know.

Reply:
To be clear, I think having 2 keys is totally reasonable because the APIs are very different. I meant to DRY out the code - possibly something like
```
function findMatchingApi(apis, apiName) {
    if (!apis || apis.items.length === 0) {
        throw new Error(`Could not find any Apis: ${JSON.stringify(apis, null, 2)}`)
    }

    const matchingApis = apis.items.filter(api => api.name === key)
    if (matchingApis.length !== 1) {
      throw new Error(`Could not find any Api with name ${key}`)
    }

    return matchingApis[0]
}
```